src/client/scripts/esm/chess/logic/movesets.ts

```diff
@@ -9,28 +9,50 @@
 
 // Type definitions...
 
+import type { gamefile } from './gamefile.js';
+
 /**
  * A Movesets object containing the movesets for every piece type in a game
- * @typedef {Object<string, PieceMoveset>} Movesets
  */
+interface Movesets {
+	[key: string]: PieceMoveset
+};
+
+/**
+ * The return value of the ignore function
+ */
+interface IgnoreResult {
+    legal: boolean,
+    blocking: boolean
+}
+
+// eslint-disable-next-line no-unused-vars
+type IgnoreFunction = (distance: number, gamefile: gamefile, detectCheck: (gamefile: gamefile, color: string, attackers: {
+	coords: number[],
+	slidingCheck: boolean
+}) => boolean) => IgnoreResult;
 
 /**
  * A moveset for an single piece type in a game
- * @typedef {Object} PieceMoveset
- * @property {number[][]} individual - Array of individual moves.
- * @property {Object<string, number[]>} [sliding] - Optional sliding moves.
  */
+interface PieceMoveset {
```

Comment:
All of these are optional

Reply:
Even individual moves? It looks like it should be required.